### PR TITLE
Adds CodeQL query to check for Pages with disabled built-in validation

### PR DESCRIPTION
**Pull request by cldrn**
Request validation is a feature in ASP.NET that protects web applications against potentially malicious content in requests, specifically against cross-site scripting attacks.

This PR includes a CodeQL query to detect applications with pages with disabled validation:
* https://docs.microsoft.com/en-us/previous-versions/aspnet/hh882339(v=vs.110)?redirectedfrom=MSDN
* https://docs.microsoft.com/en-us/previous-versions/aspnet/debza5t0(v=vs.100)?redirectedfrom=MSDN"

### CODE REVIEW

**Comment by calumgrant** [comment]
Hi Paulino, Many thanks for your contribution. As you may have seen, I have added some more commits to your branch in order to get this submission production ready. In particular, I have added some tests.

There will be a merge conflict with #2356, and the language tests will fail until #2356 is merged, but this is as expected.

**Comment by cldrn** [comment]
Thanks. I will wait for the others commits.

**Comment by jf205** [comment]
Hi @cldrn. I'll have a quick look at the docs this morning.

File: csharp/ql/src/Security Features/CWE-016/ASPNetPagesValidateRequest.qhelp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		<p>
+			The following example shows the <code>validateRequest</code> flag set to <code>false</code>
+			in a <code>Web.config</code> file for ASP.NET. This will disable validation, and leave
+			the the web application vulnerable against common XSS attacks:
+		</p>
+
+		<sample src="ASPNetPagesValidateRequestBad.config" />
+
+		<p>
+			If validateRequest is set to <code>true</code>, validation is enabled:
```
Comment (jf205) [inline csharp/ql/src/Security Features/CWE-016/ASPNetPagesValidateRequest.qhelp]:
```suggestion
			If <code>validateRequest</code> is set to <code>true</code>, validation is enabled:
```

File: change-notes/1.24/analysis-csharp.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Improvements to C# analysis
+
+The following changes in version 1.24 affect C# analysis in all applications.
+
+## New queries
+
+| **Query**                   | **Tags**  | **Purpose**                                                        |
+|-----------------------------|-----------|--------------------------------------------------------------------|
+| Page request validation is disabled (`cs/web/request-validation-disabled`) | security, external/cwe/cwe-016 | Finds where ASP.NET page request validation has been disabled, which could makes the application less secure. |
```
Comment (jf205) [inline change-notes/1.24/analysis-csharp.md]:
```suggestion
| Page request validation is disabled (`cs/web/request-validation-disabled`) | security, frameworks/asp.net, external/cwe/cwe-016 | Finds where ASP.NET page request validation has been disabled, which could makes the application less secure. |
```

File: csharp/ql/src/Security Features/CWE-016/ASPNetPagesValidateRequest.qhelp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			</code>
+		</p>
+
+	</recommendation>
+	<example>
+
+		<p>
+			The following example shows the <code>validateRequest</code> flag set to <code>false</code>
+			in a <code>Web.config</code> file for ASP.NET. This will disable validation, and leave
+			the the web application vulnerable against common XSS attacks:
```
Comment (jf205) [inline csharp/ql/src/Security Features/CWE-016/ASPNetPagesValidateRequest.qhelp]:
```suggestion
			the web application vulnerable against common XSS attacks:
```

**Comment by cldrn** [comment]
Fixed the doc as suggested.

**Comment by calumgrant** [comment]
Many thanks, I've fixed the merge conflict on the change notes.

**Pull request merged**